src/navigation/reroute.js

```diff
@@ -49,12 +49,16 @@ export function reroute(pendingPromises = [], eventArguments) {
         wasNoOp = false;
       }
 
-      return Promise.all(loadPromises)
-        .then(finishUpAndReturn)
-        .catch(err => {
-          callAllEventListeners();
-          throw err;
-        });
+      return (
+        Promise.all(loadPromises)
+          .then(callAllEventListeners)
+          // there are no mounted apps, before start() is called, so we always return []
+          .then(() => [])
```

Comment:
`triggerAppChange()` before `start()` needs to return an array of mounted apps. This used to be handled by `finishUpAndReturn`, but we're no longer calling that function for reasons described above.